meta/classes/create-spdx-2.2.bbclass

```diff
@@ -1031,6 +1031,8 @@ def combine_spdx(d, rootfs_name, rootfs_deploydir, rootfs_spdxid, packages, spdx
 
     for name in sorted(packages.keys()):
         if name not in providers:
+            if d.getVar('BUILD_IMAGES_FROM_FEEDS') == "1":
```

Comment:
We should perhaps also make sure that the package is in `IMAGE_INSTALL_NODEPS` before skipping.

Reply:
Makes sense. I'll give that a shot.

Reply:
No joy. The issue is that the proprietary package information in OE is only the top-level of the dependency tree from the ni-feeds. The nifeeds packages can and do pull in their own dependencies, which are only known to opkg. So we don't have the necessary information to build a full map of what the NODEPS packages are. As I recall, we also thought through this problem when Charlie wrote this original implementation.

Reply:
More context from offline conversation:`NI_PROPRIETARY_*` packages have dependencies on other `ni-feeds` packages (in addition to OE dependencies) which also don't have SPDX.